### PR TITLE
[Merged by Bors] - Enable logging

### PR DESCRIPTION
**Pull request by maltesander**
# Description

fixes https://github.com/stackabletech/opa-operator/issues/410

## TODO:

- [x] Release operator-rs and switch to tag (for vector config)

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by siegfriedweber — CHANGES_REQUESTED** [review]
Some remarks and the changelog must be updated.

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -52,6 +56,10 @@ pub struct OpaSpec {
     pub stopped: Option<bool>,
     /// The OPA image to use
     pub image: ProductImage,
+    /// Name of the Vector aggregator discovery ConfigMap.
+    /// It must contain the key `ADDRESS` with the address of the Vector aggregator.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub vector_aggregator_config_map_name: Option<String>,
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/crd/src/lib.rs:66]: https://github.com/stackabletech/opa-operator/pull/420/commits/2edbb77697268031ca369de617ff3e55e66b8747 and tests & docs https://github.com/stackabletech/opa-operator/pull/420/commits/cbf6ccd1a1566416589f923b71b8abd18d705961

File: rust/operator-binary/src/controller.rs [hunk 2]
```diff
@@ -435,20 +502,70 @@ fn build_server_rolegroup_daemonset(
...
         .collect::<Vec<_>>();
-    let container_opa = ContainerBuilder::new("opa")
-        .expect("invalid hard-coded container name")
+
+    let prepare_container_name = Container::Prepare.to_string();
+    let mut cb_prepare = ContainerBuilder::new(&prepare_container_name).with_context(|_| {
+        IllegalContainerNameSnafu {
+            container_name: prepare_container_name.to_string(),
+        }
+    })?;
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs]: Good point: https://github.com/stackabletech/opa-operator/pull/420/commits/8dd397d5d67589077e148ba220dd1ff622aa7c7f

File: rust/operator-binary/src/controller.rs [hunk 3]
```diff
@@ -485,32 +612,63 @@ fn build_server_rolegroup_daemonset(
...
+    .add_container(cb_bundle_builder.build())
+    .image_pull_secrets_from_product_image(resolved_product_image)
+    .add_volume(Volume {
+        name: "config".to_string(),
+        config_map: Some(ConfigMapVolumeSource {
+            name: Some(rolegroup_ref.object_name()),
+            ..ConfigMapVolumeSource::default()
+        }),
+        ..Volume::default()
+    })
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs]: Done https://github.com/stackabletech/opa-operator/pull/420/commits/865d1b1ee6947fb011e7796798ef67405c23c745

File: rust/operator-binary/src/controller.rs [hunk 4]
```diff
@@ -435,20 +502,70 @@ fn build_server_rolegroup_daemonset(
...
-        .command(build_opa_start_command())
-        .add_env_vars(env)
-        .add_container_port(APP_PORT_NAME, APP_PORT.into())
-        .add_volume_mount("config", "/stackable/config")
-        .resources(merged_config.resources.to_owned().into())
+        .command(vec![
+            "bash".to_string(),
+            "-euo".to_string(),
+            "pipefail".to_string(),
+            "-x".to_string(),
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs]: Done https://github.com/stackabletech/opa-operator/pull/420/commits/8da3335f4edbc107fccd716c88bd22a64ac6dd06

**Comment by maltesander** [comment]
test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/opa-operator-it-custom/40/

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**